pkg/process/handler.go

```diff
@@ -0,0 +1,174 @@
+// @Tags patient
+// @Description return a registration details by its ID
+// @Param RegistrationID path uint true "registration's ID"
+// @Produce json
+// @Success 200 {object} api.ReturnedData{data=Registration}
+// @Router /patient/registration/{RegistrationID} [GET]
+func (h *RegistrationHandler) GetRegistrationByPatient(c echo.Context) error {
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+
+// GetRegistrationByDoctor
+// @Summary get a registration by its ID (doctor view)
+// @Tags doctor
+// @Description return a registration details by its ID
+// @Produce json
+// @Success 200 {object} api.ReturnedData{data=Registration}
+// @Router /doctor/registration/{RegistrationID} [GET]
+func (h *RegistrationHandler) GetRegistrationByDoctor(c echo.Context) error {
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+
+// UpdateRegistrationStatus
+// @Summary update registration status
+// @Tags both doctor & patient
+// @Description update registration status
+// @Param RegistrationID body uint true "registration ID"
+// @Param Status body string true "next status of current registration"
+// @Param TerminatedCause body string false "if registration is shutdown, a cause is required"
+// @Produce json
+// @Success 200 {object} api.ReturnedData{}
+// @Router /registration [PUT]
+func (h *RegistrationHandler) UpdateRegistrationStatus(c echo.Context) error {
+	//verify identity
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+
+// CreateMileStoneByDoctor
+// @Summary create milestone
+// @Tags doctor
+// @Description the doctor create milestone (type: array)
+// @Param RegistrationID body uint true "registration's ID"
+// @Param Activity body string true "milestone's activity"
+// @Produce json
+// @Success 200 {string} api.ReturnedData{}
+// @Router /milestone [POST]
+func (h *RegistrationHandler) CreateMileStoneByDoctor(c echo.Context) error {
+	//verify identity
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+
+
+// UpdateMileStoneByDoctor
+// @Summary update milestone
+// @Tags doctor
+// @Description the doctor update milestone (check milestone)
+// @Param MileStoneID body uint true "milestone's ID"
+// @Param Activity body string false "updated milestone's activity"
+// @Param Checked body boolean true "milestone is checked or not"
+// @Produce json
+// @Success 200 {string} api.ReturnedData{}
+// @Router /milestone [PUT]
+func (h *RegistrationHandler) UpdateMileStoneByDoctor(c echo.Context) error {
+	//verify identity
+	c.Logger().Debug("hello world")
+	return c.JSON(http.StatusCreated, api.Return("ok", nil))
+}
+
+// DeleteMileStoneByDoctor
+// @Summary delete milestone
+// @Tags doctor
+// @Description the doctor update milestone (check milestone)
+// @Param MileStoneID body uint true "milestone's ID"
```

Comment:
```suggestion
// @Param MileStoneID path uint true "milestone's ID"
```

Reply:
This is a DELETE method. Will parameter in URL confuse developers?

Reply:
That's REST 😆, and that's how we do it. You refer to some resource with a URL and describe your operation with HTTP METHOD.

Reply:
So it will be:
- What is the resource? `/milestone/{MileStoneID}`
- What do we do to it? `Delete it`

Reply:
ok. I will fix it ASAP